### PR TITLE
add rotate animation

### PR DESCRIPTION
**Pull request by DominikReszczynski**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
Create a card flip interaction, after clicking (desktop) or taping (mobile) on the card it changes sides.

### Requirements
1. On tap (mobile) / click (desktop) the card should flip (change) to the other side.
2. Clicking on a card should change sides, and based on that, display a specific content.

### Code Review

#### File: src/components/AppCard/normal/normalCard.module.css

The CSS changes seem to be in line with the task requirements. The `rotate` animation has been added to create a flip effect. The `.card.flip` class applies this animation, and the opacity of the `.editContainer` and `.text` is set to 0 when the card is flipped, which presumably hides the content on one side of the card during the flip.

#### File: src/components/AppCard/normal/normalCard.tsx

The changes in this file also align with the task requirements. A new state variable `flip` has been introduced to control whether the card is flipped or not. The `useEffect` hook is used to reset the `flip` state after 400ms, which should be after the flip animation completes.

The `onClick` handler for the card button has been updated to set `flip` to true and log its value. This should trigger the flip animation and change the side of the card.

However, there's a minor issue with the `onClick` handler. The expressions inside the handler are separated by commas, which means they will all execute regardless of the condition for the ternary operator. This could lead to unexpected behavior if `isEdit` is true. To fix this, these expressions should be wrapped in a block and controlled by an if statement:

```diff
- onClick={() => {(!isEdit ? changeSide() : null), setFlip(true), console.log(`flip: ${flip}`)}}>
+ onClick={() => {
+   if (!isEdit) {
+     changeSide();
+     setFlip(true);
+     console.log(`flip: ${flip}`);
+   }
+ }}>
```

Overall, the code changes seem to cover the task requirements. The card should flip when clicked or tapped, and the content should change based on the side of the card. However, it's not clear from the provided code how the specific content for each side is determined. This would need to be verified during testing.

**Review by NerdbordMentor — APPROVED** [review]
# Code Review Report

## Task Goal
Create a card flip interaction, after clicking (desktop) or taping (mobile) on the card it changes sides.

## Requirements
- On tap (mobile) / click (desktop) the card should flip (change) to the other side.
- Clicking on a card should change sides, and based on that, display a specific content.

## Code Changes Review

### File: src/components/AppCard/normal/normalCard.module.css
The changes in this file seem to be in line with the requirements. The CSS animation for the card flip has been added. The `.card.flip` class applies a rotation animation, which gives the effect of the card flipping. The opacity of `.editConteiner` and `.text` is set to 0 when the card is flipped, which hides the content on the original side of the card during the flip.

### File: src/components/AppCard/normal/normalCard.tsx
The changes in this file also align with the requirements. A new state variable `flip` has been introduced to track whether the card is in the process of flipping. The `flip` state is set to `true` when the card is clicked, triggering the flip animation. After 400ms (the duration of the flip animation), `flip` is set back to `false`, ending the animation and revealing the content on the other side of the card.

In summary, the provided code changes appear to cover the task requirements. The card flip interaction has been implemented and clicking on a card changes sides and displays specific content based on which side is currently visible.
